lms/djangoapps/grades/tasks.py

```diff
@@ -16,8 +17,8 @@
 from .new.subsection_grade import SubsectionGradeFactory
 
 
-@task()
-def recalculate_subsection_grade(user_id, course_id, usage_id):
+@task(bind=True, default_retry_delay=30)
```

Comment:
Rather than needing to explicitly have try-except blocks and explicitly retry again, the django task framework automatically does this for you with the appropriate decorators.  Please take a look at how this is handled in the block_structure code: https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/block_structure/tasks.py#L14-L17.  Unless there's a good reason, we should be consistent with how it's handled there.